### PR TITLE
Fixed incorrect generation of union queries

### LINKED ISSUE #44: sql generated for union queries is incorrect
When doing a UNION query, Laravel wraps each union in (). SingleStore doesn't support () around unions. Need to test this and see what exactly is the issue.

### PR DESCRIPTION
**Pull request by AdalbertMemSQL**
Changed to add SELECT * FROM before each union part. 
Changed to wrap the query before adding a final order by/limit/offset for union queries. 
This is needed because otherwise these clauses are applied only to the last union subquery.

closes #44

### CODE REVIEW

File: tests/Hybrid/UnionTest.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            ['id' => 1],
+            ['id' => 2],
+            ['id' => 3],
+            ['id' => 4],
+            ['id' => 100],
+        ]);
+
+        $first = DB::table('test')->where('id', '<', 3);
+        $second = DB::table('test')->where('id', '>', 5);
+        $res = $first->union($second)->orderBy('id')->limit(1)->offset(1)->get();
```
Comment (carlsverre) [inline tests/Hybrid/UnionTest.php:63]: can you add tests which only specify one of: order, limit, offset

File: tests/Hybrid/UnionTest.php [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            ['id' => 1],
+            ['id' => 2],
+            ['id' => 3],
+            ['id' => 4],
+            ['id' => 100],
+        ]);
+
+        $first = DB::table('test')->where('id', '<', 3);
+        $second = DB::table('test')->where('id', '>', 5);
+        $res = $first->union($second)->get();
```
Comment (carlsverre) [inline tests/Hybrid/UnionTest.php:33]: might be nice to also test unionAll

**Review by carlsverre — COMMENTED** [review]
This looks awesome. Just some q's on the tests

File: src/Query/Grammar.php [hunk 3]
```diff
@@ -115,4 +115,104 @@ protected function wrapJsonFieldAndPath($column)
...
+    private function compileOffsetWithLimit($offset, $limit): string
+    {
+        // OFFSET is not valid without LIMIT
+        // Add a huge LIMIT clause
+        if (! isset($limit)) {
+            // 9223372036854775807 - max 64-bit integer
+            return ' LIMIT 9223372036854775807 OFFSET '.(int) $offset;
+        }
+
+        return ' OFFSET '.(int) $offset;
```
Comment (carlsverre) [inline src/Query/Grammar.php:216]: is explicit casting offset to an int the same way that other compilers are implemented? For example, how compileLimit works?

**Review by carlsverre — APPROVED** [review]

File: src/Query/Grammar.php (hunk 3 above)
Comment (AdalbertMemSQL) [reply src/Query/Grammar.php:216]: Yea, they are implemented in the same way 
```
    /**
     * Compile the "limit" portions of the query.
     *
     * @param  \Illuminate\Database\Query\Builder  $query
     * @param  int  $limit
     * @return string
     */
    protected function compileLimit(Builder $query, $limit)
    {
        return 'limit '.(int) $limit;
    }
```

**Pull request merged**